jmx_prometheus_javaagent/src/main/java/io/prometheus/jmx/JavaAgent.java

```diff
@@ -37,13 +38,18 @@ public static void premain(String agentArgument, Instrumentation instrumentation
      new JmxCollector(new File(file)).register();
      DefaultExports.initialize();
 
-     server = new Server(socket);
+     server = new Server();
      QueuedThreadPool pool = new QueuedThreadPool();
      pool.setDaemon(true);
      pool.setMaxThreads(10);
      pool.setMaxQueued(10);
      pool.setName("jmx_exporter");
      server.setThreadPool(pool);
+     SelectChannelConnector connector = new SelectChannelConnector();
+     connector.setHost(socket.getHostName());
```

Comment:
The code is cribbed from the `InetSocketAddress` contructor in Jetty's `Server`. So I suspect any susprising behaviour is at least consistent with what we have now.